### DISCUSSION TITLE
Fastify project in typescript gets directly into static sites

### DISCUSSION THREAD

**Discussion by iAkashPattnaik** [vercel/community#270] [Help] [answered]
I have a project in typescript based on fastify.
I need to build it into `build/` directory and use that. I do use `vercel.json`. It works fine if I use plain javascript, but in typescript, it doesn't seem to work.

<kbd>vercel.json</kbd>
```json
{
    "version": 2,
    "builds": [
        {
            "src": "./index.ts",
            "use": "@vercel/node"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "/"
        }
    ]
}
```

**Comment by samcx** [comment, marked as answer]
Hi there,

We don't recommend hosting a Node.js server on Vercel. [Vercel is committed to hosting dynamic websites](https://vercel.com/docs) with great support for [Edge](https://vercel.com/docs/concepts/functions/edge-functions) and [Serverless](https://vercel.com/docs/concepts/functions/serverless-functions).

Let us know if you have any further questions or concerns.

**Reply by iAkashPattnaik** [reply]
okay.
